### DISCUSSION TITLE
深入解析 ObjC 中方法的结构 · /method-struct

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#295] [Announcements]
https://draveness.me/method-struct

**Comment by hitner** [comment]
我测试了NSObject、NSURL都正常，
NSURL class 0x10b710fa8 meta class:0x10b710fd0
class_rw_t 0x60400006d0c0
class_ro_t 0x10b7055c0

UIView的 class 0x10ce07ff8 meta class :0x10ce08160
class_rw_t  0x60400006f180
class_ro_t  0x604000097930
为何UIView的class_ro_t不在dylib里面，不知作者有没有发现这个现象？UIKit运行时才生成class_ro_t对象吗？

**Comment by meshangqingchen** [comment]
断点怎么打印的嗯? 抽出了runtime的源码吗?

**Comment by draveness** [comment]
> @meshangqingchen
> 断点怎么打印的嗯? 抽出了runtime的源码吗?

是的

**Comment by chipengliu** [comment]
> @draveness
> > @meshangqingchen
> > 断点怎么打印的嗯? 抽出了runtime的源码吗?
> 
> 是的

请问一下，大概怎么操作呢？clone runtime源码自己编译，替换系统的runtime dylib吗？

**Comment by zhiyuzhao** [comment]
objc_msgsend 有源码，可是是汇编
https://opensource.apple.com/source/objc4/objc4-723/runtime/Messengers.subproj/objc-msg-arm64.s.auto.html
